p2p/net/swarm/swarm.go

```diff
@@ -82,6 +90,8 @@ func NewSwarm(ctx context.Context, listenAddrs []ma.Multiaddr,
 	s.cg.SetTeardown(s.teardown)
 	s.SetConnHandler(nil) // make sure to setup our own conn handler.
 
+	prom.MustRegisterOrGet(peersTotal)
```

Comment:
>  "total number of peers over all swarms", which is a good simple metric to get started with.

What would this metric be used for?

> If the swarm itself is an interesting dimension of the peers count,

It's not just an interesting dimension. not conditioning on the swarm is incorrect.

not sure you see what i mean: there's no relationship implied between two different swarms in the same process (they will most likely belong to different nodes, but not necessarily), and it will double count any connections, between swarms in the same process, and from the multiple swarms to the same peer. (This happens all the time in tests, for example). so, in an in-proc network of 5 nodes, you'd see "25 Number of connected peers" which is an incorrect statement, because there's "25 individual connections, across only 5 peers".

I don't want to merge things to the surface area (interface) that are either (partially) incorrect or that we'll just have to change right away. anything we merge is stuff that someone might start using. I'd probably start using this metric just because it's there, but it would be misleading.

If want to merge just the vendoring of the lib with a test metric, sure let's do it, but maybe go for something else, that isn't a singleton?

> Labels should only be used with dimensions that are more or less bounded. For example, a unix process ID is a bad label.

What do you mean by bounded in this case? (cause unix pids are bounded, but not in the way you mean)